Firebase/Core/FIRApp.m

```diff
@@ -293,26 +301,39 @@ - (instancetype)initInstanceWithName:(NSString *)name options:(FIROptions *)opti
 - (BOOL)configureCore {
   [self checkExpectedBundleID];
   if (![self isAppIDValid]) {
-    if (_options.usingOptionsFromDefaultPlist && [self isDataCollectionDefaultEnabled]) {
+    if (_options.usingOptionsFromDefaultPlist && [self isDataCollectionDefaultEnabled] &&
+        NSClassFromString(@"FIRDiagnostics")) {
+      FIRDiagnosticsData *diagnosticsData = [[FIRDiagnosticsData alloc] init];
+      [diagnosticsData insertValueIfNotNil:@([self isDataCollectionDefaultEnabled])
+                                    forKey:kFIRCDIsDataCollectionDefaultEnabledKey];
+      [diagnosticsData insertValueIfNotNil:[FIRApp firebaseUserAgent]
+                                    forKey:kFIRCDFirebaseUserAgentKey];
+      [diagnosticsData insertValueIfNotNil:@(FIRConfigTypeCore) forKey:kFIRCDConfigurationTypeKey];
       [[NSNotificationCenter defaultCenter]
-          postNotificationName:kFIRAppDiagnosticsNotification
+          postNotificationName:kFIRDiagnosticsNotification
                         object:nil
-                      userInfo:@{
-                        kFIRAppDiagnosticsConfigurationTypeKey : @(FIRConfigTypeCore),
-                        kFIRAppDiagnosticsErrorKey : [FIRApp errorForInvalidAppID],
-                      }];
+                      userInfo:@{kFIRDiagnosticsDataNotifKey : diagnosticsData}];
     }
     return NO;
   }
 
-  if ([self isDataCollectionDefaultEnabled]) {
+  if ([self isDataCollectionDefaultEnabled] && NSClassFromString(@"FIRDiagnostics")) {
```

Comment:
Same comments as above - can we make FIRDiagnosticsData read all the data it needs instead of passing it in?